a2091.cpp

```diff
@@ -3628,9 +3630,11 @@ static void mbdmac_write_word (struct wd_state *wd, uae_u32 addr, uae_u32 val)
 		break;
 	case 0x5e:
 	case 0x80:
+#ifdef WITH_DSP
 		if (is_dsp_installed) {
 			dsp_write(val);
```

Comment:
an alternative to placing these ifdefs all over the place could be something like the following in a single header like uae_features.h:

```c
#ifndef WITH_DSP
#define is_dsp_installed 0
#define dsp_write(X) do{}while(0);
...
#else
#include <dspfuncs.h>
#endif
```
apart from the obvious lack of ifdef clutter, another advantage would be that the code is always evaluated by the compiler and no bitrot in rarely enabled codepaths can creep in, and dead code elimination will then just remove the if(0) blocks.

Reply:
Sure, but that would require more changes. For now, I'm just following the set examples of what's already there. If Toni (or you) wants to refactor things further, I'm all for it.